src/app/beer_garden/api/http/authentication/__init__.py

```diff
@@ -0,0 +1,72 @@
+from datetime import datetime, timedelta
+from typing import Optional
+
+import jwt
+from marshmallow import Schema, fields
+from tornado.httputil import HTTPServerRequest
+
+from beer_garden import config
+from beer_garden.api.http.authentication.login_handlers import LOGIN_HANDLERS
+from beer_garden.authorization import permissions_for_user
+from beer_garden.db.mongo.models import User
+
+
+class LoginInputSchema(Schema):
+    """Schema for the user login input"""
+
+    username = fields.Str()
+    password = fields.Str()
+
+
+class LoginResponseSchema(Schema):
+    """Schema for the user login response"""
+
+    token = fields.Str()
+    message = fields.Str()
+
+
+def user_login(request: HTTPServerRequest) -> Optional[User]:
+    """Attempt to authenticate a user based on the information supplied in the request.
+    Each handler from beer_garden.api.http.authentication.login_handlers will attempt
+    to authenicate the user. Once a successful authentication occurs, no subsequent
+    handlers will be tried. If none of the handlers is able to authenticate, the login
+    attempt will be denied and None will be returned.
+
+    Args:
+        request: The tornado HTTPServerRequest containing login information.
+
+    Returns:
+        User: The User object for the successfully authenticated user
+        None: If unable to authenticate a user based on the request
+    """
+    user = None
+
+    for handler in LOGIN_HANDLERS:
+        user = handler.get_user(request)
+
+        if user:
+            break
+
+    return user
+
+
+def generate_access_token(user: User) -> str:
+    """Generates a JWT access token for a user containing the user's permissions
+
+    Args:
+      user: The User to generate the access token for
+
+    Returns:
+      str: The encoded JWT
+    """
+    secret_key = config.get("auth").token.secret
+
+    jwt_headers = {"alg": "HS256", "typ": "JWT"}
```

Comment:
Yeah, I think removing the configuration option is the right thing to do.  For this particular use case, I don't think it makes sense to have the expiration be configurable.  I'll look at taking the configuration parameter away and hopefully none of the old (unused) token stuff is in the critical path somewhere such that the config not existing would break things.

Reply:
I removed the `algorithm` and `lifetime` settings, which left just the `secret` setting, so I decided to just push that up to the `auth` level as `token_secret` rather than having `token` be its own subsection of a single config.